### DISCUSSION TITLE
"Error loading data Unable to render chart." in custom charts

### DISCUSSION THREAD

**Discussion by polamjag** [community/community#47518] [Projects and Issues]
### Select Topic Area

Bug

### Body

Steps to reproduce:

- Set chart's Y-axis to "Sum of a field" of some numeric field
  - X-axis is Time
  - Changing layout doesn't matter

<img width="1028" alt="scr 2023-02-15 at 15 32 48" src="https://user-images.githubusercontent.com/685936/218950451-87ec6367-4cb7-4a4e-8345-72b19134ebe6.png">

**Comment by ryders** [comment]
I have had the same problem for a few months now, with no avenue to getting any help from github.

from my early investigation, it seems this only happens on charts with **Time** as X-Axis.
![Screenshot 2023-02-20 at 1 56 01 am](https://user-images.githubusercontent.com/242383/219986991-a19d20b9-ecb4-43b0-b2e6-727590b01fa5.png)
![Screenshot 2023-02-20 at 1 56 10 am](https://user-images.githubusercontent.com/242383/219987001-070c9a54-4a70-4b3c-b599-ae449ba2e847.png)
![github-chart](https://user-images.githubusercontent.com/242383/219987009-039bdd40-475c-4c5c-b9d7-1fd36f34ff82.gif)

**Comment by chiedo** [comment]
Thanks for reporting this folks! So far I haven't been able to reproduce but I'm going to share this with the team.

**Reply by beingminimal** [reply]
@chiedo how to get access for this time feature in project insights? I have access to one org in github and now our team required it for another org in github please help. Thanks

**Reply by rileybroughten** [reply]
@beingminimal historical (time-based) charts are available for Enterprise Cloud accounts, and in a limited preview for Team accounts. No opt-in is necessary and if have access, you will see historical charts available. Documentation on that is [here](https://docs.github.com/en/issues/planning-and-tracking-with-projects/viewing-insights-from-your-project/about-insights-for-projects) 👍

**Comment by polamjag** [comment]
After some investigation with DevTools I found that POST request is sent to `https://github.com/memexes/:id/insights/query?period=1M` with payload like this, maybe in order to obtain data for chart:

```
{
  "query": "--render stacked-area-chart-memex-insights\n",
  "queryScope": null,
  "options": {
    "includeArchived": false,
    "aggregate": {
      "column": "ポイント",
      "aggregationType": "sum"
    },
    "groupBy": {
      "column": "Status",
      "dataType": "singleSelect"
    },
    "projectItemIds": [ <snip> ]
  }
}
```

results in 400 Bad Request with message `{"errors":["aggregate column is invalid"]}`.

I hope this helps in your investigation :)

**Comment by ryders** [comment]
@chiedo further to this -- I don't know whether this is linked, but it's certainly related.

I use the same chart in a different project, which renders fine. it that other project, I can also display "story points" in a board view and it shows up in the column heading.

In my main project (with the broken charts as reported above), these do not show.

So clearly -- there's a problem around "story points".
![Screenshot 2023-03-05 at 11 28 30 pm](https://user-images.githubusercontent.com/242383/222989793-bfbdc347-2b1d-48f3-8854-06e74b117e9b.png)
![Screenshot 2023-03-05 at 11 28 59 pm](https://user-images.githubusercontent.com/242383/222989795-23611a09-78bf-4d34-9078-5ee4af3df0ad.png)

**Reply by rileybroughten** [reply]
@ryders do you see the option for `field sum` in the same menu? That is the control to toggle and display the sum for numeric fields on your project and board columns in this case. The `fields` menu controls which fields are display on the board items themselves.

<img width="605" alt="Screenshot 2023-04-03 at 9 19 04 AM" src="https://user-images.githubusercontent.com/101840513/229568746-62377567-d47e-4fc3-9bbc-10978f1813e7.png">

**Reply by ryders** [reply]
Hi @rileybroughten -- yup, the field sum for Story Points does show, and also does calculate fine.
![Screenshot 2023-04-03 at 21 26 36](https://user-images.githubusercontent.com/242383/229608234-3760341c-d856-491d-a9a8-b66120ce4f4e.png)

Mind boggling really! Makes me happy to realise that event products the size and maturity of Github can harbour bugs! 😆

**Reply by ryders** [reply]
Still no luck on charts tho.

One additional piece of clarification:

> I use the same chart in a different project,

That should read "in a different project **for a different organisation**". It seems a different project within the same organisation doesn't work either.

**Comment by jss-tg** [comment]
It looks like the field names are currently only supported in English.😄️
I changed "故事点" to "Point" and the chart displays correctly.

**Comment by bonty** [comment]
We have similar problem in our projects. We found this problem will reproduce when we have added multiple repository in an project.

**Comment by Cr0wn-Gh0ul** [comment]
I am also having this issue. The request fails with `{"message":"Error fetching insights data"}`

**Reply by widal001** [reply]
Are you a member of the organization that owns the repository and the GH project from which the data is being pulled? 

I saw the same error message when I was added to a repository as an external collaborator. Once the org admin added me to the organization as a member, I was able to see the burn up chart (and other historical charts).

I think it has to do with how access to historical information about a repository is controlled.

**Comment by MrChocolateMoose** [comment]
I am having this issue as well

**Comment by rileybroughten** [comment]
👋 Support for the `Group by` field in historical charts (charts with Time as X-axis, both the "Burn up" chart and any custom charts you created) is being gradually phased out on existing historical charts over the coming weeks, which may result in some disruptions in the meantime. Here is a link to the [documentation](https://docs.github.com/en/issues/planning-and-tracking-with-projects/viewing-insights-from-your-project/about-insights-for-projects) for more details, and we'll be providing a new experience for the "Burn up" chart so you can view the progress of your items over time 👍

**Comment by ianstokesrees-gamma** [comment]
@rileybroughten I've read the new documentation but it isn't clear how my teams will create "burn up" (and of course probably like others we'd prefer "burn-down") charts in the new system. It seems surprising that GH would remove a key feature *before* having a clear replacement in place, so what am I missing?

If it helps, I'm part of the BCG Github EMU environment, and the specific organization (managed within that space) that has this issue is https://github.com/orgs/bcgx-pi-472441-19

**Reply by rileybroughten** [reply]
We will be providing a new "Burn up" chart experience so that you can see the progress and state of your issues and pull requests over time. We don't have a timeline yet of when we'll be providing a burn down chart, but would keep an eye on the [GitHub public roadmap](https://github.com/orgs/github/projects/4247/views/7) for future updates. We will have the errors resolved within the next couple weeks and apologize for the disruptions in the meantime!

**Reply by beingminimal** [reply]
@rileybroughten when github pro users will get access for time based historical chart support?

**Comment by rainbow1016** [comment]
very good question!
